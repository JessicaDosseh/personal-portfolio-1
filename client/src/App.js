import React, { useState } from "react";
import { GlobalStyle } from "./components/styles";
import { Route } from "react-router-dom";
import styled, { css } from "styled-components";
import PropTypes from "prop-types";
import { NavbarContainer } from "./components/NavBar";
import { HomePageContainer } from "./components/HomePage";
import { ContactForm } from "./components/ContactForm";
import { EmailModal } from "./components/EmailModal";
import { SendingModal } from "./components/SendingModal";
import { InvalidEmailModal } from "./components/InvalidEmailModal";

const AppDiv = styled.div`
  width: 100%;
  min-height: 100vh;
  background: white;
  padding: 0 5%;
  position: relative;

  ${props =>
    props.darkmode &&
    css`
      background: #1b2331;
    `}

  ${props =>
    (props.displayModal || props.sending) &&
    css`
      overflow: hidden;
      position: fixed;
    `}
`;

const App = props => {
  const pullMode = localStorage.getItem("darkMode") === "true" ? true : false;
  const [darkmode, setDarkmode] = useState(pullMode);
  const [invalidEmailModal, setInvalidEmailModal] = useState(false);
  const [emailValues, setEmailValues] = useState({
    sending: false,
    success: props.null,
    displayModal: false,
    modalText: ""
  });

  return (
    <>
      <GlobalStyle />
      <AppDiv
        darkmode={darkmode}
        sending={emailValues.sending}
        displayModal={emailValues.displayModal}
      >
        {emailValues.displayModal && (
          <EmailModal
            emailValues={emailValues}
            setEmailValues={setEmailValues}
            darkmode={darkmode}
            match={props.match}
            history={props.history}
          />
        )}
        {emailValues.sending && <SendingModal darkmode={darkmode} />}
        {invalidEmailModal && (
          <InvalidEmailModal
            setInvalidEmailModal={setInvalidEmailModal}
            darkmode={darkmode}
          />
        )}
        <div>
          <NavbarContainer
            setDarkmode={setDarkmode}
            darkmode={darkmode}
            history={props.history}
          />
          <Route
            exact
            path="/"
            render={props => (
              <HomePageContainer {...props} darkmode={darkmode} />
            )}
          />
          <Route
            path="/contact"
            render={props => (
              <ContactForm
                {...props}
                darkmode={darkmode}
                emailValues={emailValues}
                setEmailValues={setEmailValues}
                setInvalidEmailModal={setInvalidEmailModal}
              />
            )}
          />
        </div>
      </AppDiv>
    </>
  );
};

App.propTypes = {
  darkmode: PropTypes.bool,
  setDarkmode: PropTypes.func
};

export default App;
