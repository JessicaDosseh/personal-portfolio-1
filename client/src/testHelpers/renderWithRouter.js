import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
