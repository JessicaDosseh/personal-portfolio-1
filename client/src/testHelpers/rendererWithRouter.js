import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
