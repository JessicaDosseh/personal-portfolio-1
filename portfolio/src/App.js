import React, { useState } from "react";
import { GlobalStyle } from "./components/styles";
import { NavbarContainer } from "./components/NavBar";
import { HomePageContainer } from "./components/HomePage";
import { Route } from "react-router-dom";
import styled, { css } from "styled-components";

const AppDiv = styled.div`
  width: 100%;
  background: white;

  ${props =>
    props.darkmode &&
    css`
      background: #2f3234;
    `}
`;

const App = () => {
  const pullMode = localStorage.getItem("darkMode") === "true" ? true : false;
  const [darkmode, setDarkmode] = useState(pullMode);

  return (
    <>
      <GlobalStyle />
      <AppDiv darkmode={darkmode} className="App">
        <NavbarContainer setDarkmode={setDarkmode} darkmode={darkmode} />
        <Route
          exact
          path="/"
          render={props => <HomePageContainer {...props} darkmode={darkmode} />}
        />
      </AppDiv>
    </>
  );
};

export default App;
