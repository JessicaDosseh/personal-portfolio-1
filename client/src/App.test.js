import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "./testHelpers";

import App from "./App";

describe("<App />", () => {
  it("should render the App component without crashing", () => {
    const push = jest.fn();
    const scrollToSpy = jest.fn();
    window.scrollTo = scrollToSpy;
    const helpers = renderWithRouter(
      <App history={push} history={{ location: { pathname: "/" } }} />
    );
  });

  it("matches the snapshot of App", () => {
    const tree = rendererWithRouter(
      <App history={{ location: { pathname: "/" } }} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
