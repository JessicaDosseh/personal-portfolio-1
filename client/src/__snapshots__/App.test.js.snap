// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot of App 1`] = `
<div
  className="sc-cSHVUG bwEByw"
>
  <div>
    <div
      className="sc-ifAKCX iSFgrr"
    >
      <div
        className="sc-bwzfXH hyjEOv"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            className="sc-bdVaJa istZpN"
          >
            <div>
              <p>
                N
              </p>
            </div>
          </div>
        </a>
        <div
          className="sc-htpNat jYdygF"
        >
          <div
            className="sc-bxivhb NNymt"
            data-testid="darkmodeTab"
            onClick={[Function]}
          >
            <div>
              <img
                alt="Dark mode logo"
                src="moon.svg"
              />
            </div>
          </div>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Work
          </a>
          <a
            aria-current={null}
            href="/contact"
            onClick={[Function]}
          >
            Contact
          </a>
        </div>
      </div>
    </div>
    <div>
      <div
        className="sc-EHOje cHEDNC"
      >
        <p>
          I'm Nathan Thomas, a
           
          <a
            href="https://github.com/nwthomas"
          >
            full stack software engineer
          </a>
           
          living in the Napa Valley and currently attending
           
          <a
            href="https://lambdaschool.com/"
          >
            Lambda School
          </a>
          . Let's
           
          <a
            href="/contact"
            onClick={[Function]}
          >
            talk
          </a>
          .
        </p>
        <div
          className="sc-bZQynM kIarEe"
        >
          <div>
            <h4>
              Current Student:
            </h4>
            <div>
              <a
                href="https://lambdaschool.com/"
              >
                <img
                  alt="Current school logo"
                  data-testid="schoolLogo"
                  src="lambda-logo.png"
                />
              </a>
            </div>
          </div>
          <div>
            <h4>
              Previously:
            </h4>
            <div>
              <a
                href="https://www.adventisthealth.org/"
              >
                <img
                  alt="Former employer logo"
                  data-testid="workLogo"
                  src="ah-logo.png"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-gzVnrw kufNtn"
      >
        <p
          name="featuredProjects"
        >
          Featured Projects
        </p>
        <div>
          <div>
            <div>
              <a
                href="https://github.com/nwthomas/tippr"
              >
                <img
                  alt="tippr portfolio samples"
                  data-testid="tipprPicture"
                  src="tippr-macbook.png"
                />
              </a>
              <h4>
                tippr Payment App
              </h4>
            </div>
          </div>
          <div>
            <div>
              <a
                href="https://github.com/nwthomas/instagram-clone"
              >
                <img
                  alt="Instagram portfolio samples"
                  data-testid="instagramPicture"
                  src="instagram-macbook.png"
                />
              </a>
              <h4>
                Instagram Clone
              </h4>
            </div>
          </div>
          <div>
            <div>
              <a
                href="https://github.com/nwthomas/fylo-landing-page"
              >
                <img
                  alt="Fylo portfolio samples"
                  data-testid="fyloPicture"
                  src="fylo-macbook.png"
                />
              </a>
              <h4>
                Fylo Landing Page
              </h4>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-htoDjs bnNaEF"
      >
        <div>
          <p>
            Looking to start a project?
          </p>
          <a
            href="/contact"
            onClick={[Function]}
          >
            Get in contact.
          </a>
        </div>
        <div>
          <a
            href="https://github.com/nwthomas"
          >
            GitHub
          </a>
          <a
            href="https://www.linkedin.com/in/nwthomas-profile/"
          >
            LinkedIn
          </a>
          <a
            href="https://medium.com/@nwthomas"
          >
            Medium
          </a>
          <a
            href="https://twitter.com/nwthomas_"
          >
            Twitter
          </a>
          <a
            href="https://www.instagram.com/nwthomas/"
          >
            Instagram
          </a>
          <a
            href="/contact"
            onClick={[Function]}
          >
            Email
          </a>
        </div>
      </div>
      <div
        className="sc-dnqmqq dMVQnI"
      >
        <h4>
          Copyright Â© 2019 Nathan Thomas
        </h4>
      </div>
    </div>
  </div>
</div>
`;
